test/quantization/pt2e/test_numeric_debugger.py

```diff
@@ -128,29 +160,67 @@ def test_re_export_preserve_handle(self):
         generate_numeric_debug_handle(ep)
         m = ep.module()
 
-        debug_handle_map_ref = _extract_debug_handles(m)
-        m_export = export_for_training(m, example_inputs).module()
-        debug_handle_map = _extract_debug_handles(m_export)
+        self._assert_each_node_has_debug_handle(ep)
+        debug_handle_map_ref = self._extract_debug_handles(ep)
+
+        ep_reexport = export_for_training(m, example_inputs)
+
+        self._assert_each_node_has_debug_handle(ep_reexport)
+        debug_handle_map = self._extract_debug_handles(ep_reexport)
 
         self.assertEqual(debug_handle_map, debug_handle_map_ref)
 
-    def test_run_decompositions_preserve_handle(self):
+    def test_run_decompositions_same_handle_id(self):
         m = TestHelperModules.Conv2dThenConv1d()
         example_inputs = m.example_inputs()
         ep = export_for_training(m, example_inputs)
         generate_numeric_debug_handle(ep)
 
-        debug_handle_map_ref = _extract_debug_handles(ep)
+        self._assert_each_node_has_debug_handle(ep)
+        debug_handle_map_ref = self._extract_debug_handles(ep)
 
         ep_copy = copy.copy(ep)
         ep_copy = ep_copy.run_decompositions()
-        debug_handle_map = _extract_debug_handles(ep_copy)
+
+        self._assert_each_node_has_debug_handle(ep_copy)
+        debug_handle_map = self._extract_debug_handles(ep_copy)
 
         # checking the map still has the same ids, the node may change
         self.assertEqual(
             set(debug_handle_map.values()), set(debug_handle_map_ref.values())
         )
 
+    def test_run_decompositions_map_handle_to_new_nodes(self):
+        m = TestHelperModules.TwoLinearModule()
+
+        test_models = [
+            TestHelperModules.TwoLinearModule(),
+            TestHelperModules.Conv2dThenConv1d(),
+            TestHelperModules.ControlFlow(),
+        ]
+
+        for m in test_models:
+            example_inputs = m.example_inputs()
+            ep = export_for_training(m, example_inputs)
+            generate_numeric_debug_handle(ep)
+
+            self._assert_each_node_has_debug_handle(ep)
+            pre_decomp_to_debug_handle_map_ref = (
+                self._extract_debug_handles_with_prev_decomp_op(ep)
+            )
+
+            ep_copy = copy.copy(ep)
+            ep_copy = ep_copy.run_decompositions()
+            self._assert_each_node_has_debug_handle(ep_copy)
+            pre_decomp_to_debug_handle_map = (
```

Comment:
you mean `post`?

Reply:
It is `pre`. The map key is the pre-decompose operator name, and value is the debug handle in the post debug handle. Maybe I need to have a better name haha

Reply:
oh I see, makes sense